include/odp/api/spec/packet_io.h

```diff
@@ -106,6 +106,10 @@ typedef enum odp_pktout_mode_t {
 	ODP_PKTOUT_MODE_QUEUE,
 	/** Packet output through traffic manager API */
 	ODP_PKTOUT_MODE_TM,
+	/** Packet output through traffic manager API where
+	 * TM queues act as event queues.
+	 */
+	ODP_PKTOUT_MODE_TM_QUEUE,
```

Comment:
I thought that the current behavior already was order-preserving as the linux-gen implementation appears to be like that. But maybe there are use case for non-order preserving enqueue too so that two modes here are useful.

Reply:
Yes, it is equivalent to Direct mode and Queue mode in non-TM PKTIO.

Reply:
As proposed by Petri, Let's not have a new pktout mode. From pktout point-of-view it is sufficient to tell that TM will be used.

Whether input queue order is preserved in TM output (the same way as in ODP queue enqueue) can be configured through the TM API. We discussed that such config could be part of odp_tm_requirements_t passed to odp_tm_create(), but now I am wondering if it should actually be part of odp_tm_queue_param_t and thus a separate (possibly differing) setting for each input queue of a TM.